SU-SWS/ansible-playbooks
# Sites 163 update servers

# READY FOR REVIEW

# Summary
- This branch includes updates to the server setup and migration playbooks, so that we can run them against our new instance.  I have tested and then ran against all servers the server-settings-playbook and tested against test-cardinalsites.acsitefactory.com a migration of: http://pursuingsustainability.test-cardinalsites.acsitefactory.com along with the 9 Earth sites.  I removed the vhost on [http://dekaslab.test-cardinalsites.acsitefactory.com](http://dekaslab.test-cardinalsites.acsitefactory.com), so this site could be viewed on ACSF.

# Needed By (10/23)

# Urgency
- I think many people are anxious for our initial pilot sites to be migrated.

# Steps to Test

*server-settings-playbook*
1. Checkout this branch.
2. Copy default.servers to inventory/servers
3. Run: `ansible-playbook -i inventory/servers server-settings-playbook.yml`
4. Log into a server or two and be sure the required drush extensions and bash customizations are present.

*migration-playbook*
1. I have migrated `pursuing-sustainability` along with the 9 Earth sites to [test-cardinalsites.acsitefactory.com](test-cardinalsites.acsitefactory.com).  If you want to migrate a site on production, you'll need to deploy tag `0.0.1`.  If you want to migrate a site to dev, you'll need to deploy the dev environment and then deploy tag `0.0.1` from there.  The code for tag `0.0.1` can be found here: https://github.com/SU-SWS/acsf-cardinald7/tree/0.0.1.
2. In migration_vars.yml, you'll need to update the `sitefactory_environment` to `test-/dev-/cardinalsites` and `acquia_api_key` value to your new key.  You'll also need to add `wait_time`.  From testing the 9 Earth sites, ACSF built 3 sites at a time and each batch took 6 minutes.  So I put 30m just to be safe.  If you're migrating more than 9 at a time, I would increase this number.
3. Double check you have the list of sites you want migrated in inventory/sites.
4. Run: `ansible-playbook -i inventory/sites migration-playbook.yml`
5. If you need to restart, run: `ansible-playbook -i inventory/sites migration-playbook.yml --tags "upload-site"`, for example.
6. Check that the sites appear as expected in the site factory environment you chose in step 1.

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?  Does it ever.

# Associated Issues and/or People
- https://stanfordits.atlassian.net/browse/SITES-167
- https://stanfordits.atlassian.net/browse/SITES-163
- https://stanfordits.atlassian.net/browse/SITES-165

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
